manifests/journald.pp

```diff
@@ -10,21 +10,23 @@
   service{'systemd-journald':
     ensure => running,
   }
-  $systemd::journald_settings.each |$option, $value| {
-    ini_setting{
-      $option:
-        path    => '/etc/systemd/journald.conf',
-        section => 'Journal',
-        setting => $option,
-        notify  => Service['systemd-journald'],
-    }
-    if $value =~ Hash {
-      Ini_setting[$option]{
-        * => $value,
+  if $systemd::journald_settings {
```

Comment:
How about using an empty hash as a default instead, so you can keep the loop but it just does nothing?

Reply:
That can be empty, this is check with `Systemd::JournaldSettings` format ?

Reply:
All the keys in that type are optional, so yes an empty hash looks legit.

Reply:
Sorry, I just not add a empty hash per default for puppet server memory, this is useless for me but we can change to an empty hash if you want

Reply:
Well it would avoid patching the loops, so I think it's much simpler.

Reply:
I'm not totally agree with that but it's done :)